drivers/serial/uart_npcx.c

```diff
@@ -90,11 +90,23 @@ static void uart_npcx_clear_rx_fifo(const struct device *dev)
 	while (uart_npcx_rx_fifo_available(dev))
 		scratch = inst->URBUF;
 }
+
+/* Forward declaration to be used in uart_npcx_poll_in(). */
+static int uart_npcx_fifo_read(const struct device *dev, uint8_t *rx_data,
+				  const int size);
+
+/* Forward declaration to be used in uart_npcx_poll_out(). */
+static int uart_npcx_fifo_fill(const struct device *dev,
+				  const uint8_t *tx_data,
+				  int size);
 #endif
 
 /* UART api functions */
 static int uart_npcx_poll_in(const struct device *dev, unsigned char *c)
 {
+#ifdef CONFIG_UART_INTERRUPT_DRIVEN
```

Comment:
One minor comment. If we replace this line with 'if (IS_ENABLED(CONFIG_UART_INTERRUPT_DRIVEN))', does it causes build-error when CONFIG_UART_INTERRUPT_DRIVEN is not set? If not, I think using 'IS_ENABLED()' macro is more readable.

Reply:
In this case it might be better to just do:
```
#ifdef CONFIG_UART_INTERRUPT_DRIVEN
static int uart_npcx_poll_in(const struct device *dev, unsigned char *c)
{
   .... /* interrupt version */
}
#else
static int uart_npcx_poll_in(const struct device *dev, unsigned char *c)
{
   .... /* poll version */
}
#endif

Reply:
@galak I think you're right, I'll just roll these into the above #ifdef.